src/libs/actions/Report.ts

```diff
@@ -2032,6 +2033,13 @@ function openReportFromDeepLink(url: string, isAuthenticated: boolean) {
                     return;
                 }
 
+                // When deep-linking to desktop app with `transition` route we don't want to call navigate
+                // on the route because it will display an infinite loading indicator.
+                // See issue: https://github.com/Expensify/App/issues/33149
+                if (getPlatform() === CONST.PLATFORM.DESKTOP && route.includes(ROUTES.TRANSITION_BETWEEN_APPS)) {
```

Comment:
we should not use platform specific code
If you want to have something just only for desktop - you can create a function openReportFromDeepLink with .desktop.ts and index.ts - and use it in this file
ex. src/libs/getPlaidDesktopMessage

Reply:
@narefyev91  Moved the logic to platform-dependent files:
- @libs/actions/OpenReportFromDeepLink/index.ts
- @libs/actions/OpenReportFromDeepLink/index.desktop.ts

they are located in `@libs/actions/...` because the `openReportFromDeepLink` function performs `Onyx.set()` which would throw lint error if placed in `@libs/...` because of the eslint rule:
- `eslint(rulesdir/prefer-actions-set-data)`

_

I also added videos for all platforms as discussed.

Reply:
Yup seems too many same logic goes in files
Can we just extract if (getPlatform() === CONST.PLATFORM.DESKTOP && route.includes(ROUTES.TRANSITION_BETWEEN_APPS)) this for a specific function? and all other code will stay at it is? @ikevin127

Reply:
@narefyev91 Sure, I think this way is better because there's no duplicated logic.
I created an action that checks if `shouldSkipDeepLinkNavigation`, platform-dependent:
- `@libs/shouldSkipDeepLinkNavigation/index.desktop.ts` (implementation logic)
- `@libs/shouldSkipDeepLinkNavigation/index.ts` (no-op, returns false)

Let me know if this needs any other changes!